### PR TITLE
[ISSUE-#8210] fix connectSync not work

### PR DESCRIPTION
**Pull request by jiangxch**
## What is the purpose of the change
fix connectSync not work

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by horizonzy** [comment]
Hi, can you expain it's intention detaily.

**Comment by horizonzy** [comment]
Ok, I have seen your issue, I will check it.

File: dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/api/Connection.java [hunk 1]
```diff
@@ -149,14 +153,16 @@ public void onGoaway(Channel channel) {
...
         channel.attr(CONNECTION).set(this);
         if (logger.isInfoEnabled()) {
             logger.info(String.format("%s connected ", this));
         }
     }
 
-    public void connectSync() {
+    public void connectSync() throws InterruptedException, ExecutionException, TimeoutException {
         this.initPromise.awaitUninterruptibly(this.connectTimeout);
+        this.onConnectFuture.get(this.connectTimeout, TimeUnit.MILLISECONDS);
```
Comment (horizonzy) [inline dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/api/Connection.java:165]: Here use CountDownLatch will be better.

Comment (horizonzy) [reply dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/api/Connection.java:165]: And here onConnectFuture is stateful, it should be reset when connect close.

File: dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/api/ConnectionTest.java [hunk 2]
```diff
@@ -55,4 +60,17 @@ public void testRefCnt2() throws InterruptedException {
...
+
+    @Test
+    public void connectSyncTest() throws Exception {
+        int port = NetUtils.getAvailablePort();
+        URL url = new ServiceConfigURL("empty", "localhost", port,
+            new String[]{Constants.BIND_PORT_KEY, String.valueOf(port)});
+        PortUnificationServer server = new PortUnificationServer(url);
+        server.bind();
+
+        Connection connection = new Connection(url);
```
Comment (horizonzy) [inline dubbo-remoting/dubbo-remoting-api/src/test/java/org/apache/dubbo/remoting/api/ConnectionTest.java:72]: @Test
    public void connectSyncTest() throws Exception {
        int port = NetUtils.getAvailablePort();
        URL url = new ServiceConfigURL("empty", "localhost", port,
                new String[]{Constants.BIND_PORT_KEY, String.valueOf(port)});
        PortUnificationServer server = new PortUnificationServer(url);
        server.bind();

        Connection connection = new Connection(url);
        connection.connectSync();
        connection.close();
        connection.connectSync();
        Assertions.assertNotNull(connection.getChannel());
    }

**Review by xiaoheng1 — COMMENTED** [review]
Similar to #8126

File: dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/api/Connection.java (hunk 1 above)
Comment (jiangxch) [reply dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/api/Connection.java:165]: yes,i will update it today

Comment (horizonzy) [reply dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/api/Connection.java:165]: Maybe say sry jiangxch, this pr is duplicated with #8126, thanks for your contribution.

Comment (jiangxch) [reply dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/api/Connection.java:165]: ok

**Pull request closed without merging**